### PR TITLE
Implement half damage to petrified units

### PR DESCRIPTION
**Pull request by vmarkovtsev**
Another TODO

### CODE REVIEW

**Comment by vmarkovtsev** [comment]
BTW, how about giving greater basilisks 25% stone gaze chance? They are greater, you know :)

**Comment by vmarkovtsev** [comment]
... and does "blinded unit retaliates" mean that currently blinded units strike back?

**Comment by ArseniyShestakov** [comment]
Yes, it's 50% damage on basic and 25% on advanced level. On expert blinded stack don't strikes back [http://heroes.thelazy.net/wiki/Blind]([1]).

**Comment by ArseniyShestakov** [comment]
About altering mechanics: you can always make it moddable later on :-)

**Comment by alexvins** [comment]
It should not be hardcoded. It should be additional spell effect.
Smth like:
```javascript
"defenceBoost":{			"type" : "PRIMARY_SKILL",
						"subtype" : "primSkill.defence",
						"val" : 50,
						"valueType" : "PERCENT_TO_ALL",
						"duration" : "N_TURNS"}
```

File: lib/CBattleCallback.cpp [hunk 1]
```diff
@@ -1020,8 +1020,12 @@ TDmgRange CBattleInfoCallback::calculateDmgRange(const BattleAttackInfo &info) c
 		multBonus *= 0.5;
 	}
 
-	// TODO attack on petrified unit 50%
-	// blinded unit retaliates
+	// attack on petrified unit 50%
+	if(info.defenderBonuses->getBonus(Selector::source(Bonus::SPELL_EFFECT, SpellID::STONE_GAZE)))
```
Comment (alexvins) [inline lib/CBattleCallback.cpp]: This is wrong on it own: stoneGaze has 2 effects with different duration.

**Comment by vmarkovtsev** [comment]
Did as @alexvins wrote

**Pull request merged**